changelog/+upgrade-deps-20251206.changed.md

```diff
@@ -0,0 +1 @@
+Upgraded backend dependencies: Neo4j to 2025.10.1, Redis to 8.4.0, RabbitMQ to 4.2.1, PostgreSQL to 18
```

Comment:
This gets a bit weird as the versions aren't strictly tied to the code and anyone could run Infrahub with some undefined version of these backend services. It's correct in terms of what we default to throughout the repo and it's probably better to include a statement of the changes to the default services. But we could consider how or where this should be presented in the future.